src/main/java/com/github/tomakehurst/wiremock/stubbing/InMemoryStubMappings.java

```diff
@@ -67,33 +64,19 @@ public ServeEvent serveFor(Request request) {
 		
 		scenarios.onStubServed(matchingMapping);
 
-        ResponseDefinition responseDefinition = applyTransformations(request,
-            matchingMapping.getResponse(),
-            ImmutableList.copyOf(transformers.values()));
+        ResponseDefinition response = matchingMapping.getResponse();
+
+		for (ResponseDefinitionTransformer transformer : response.selectApplicableResponseDefinitionTransformers(this.transformers.values())) {
+			response = transformer.transform(request, response, rootFileSource.child(FILES_ROOT), response.getTransformerParameters());
+		}
 
 		return ServeEvent.of(
             LoggedRequest.createFrom(request),
-            copyOf(responseDefinition),
+            copyOf(response),
```

Comment:
one more thing. We copy the `responseDefinition` only at the end of the transformation chain. That means that a not-so-nice transformation can mutate the state of our stub mapping, e.g.: 

```java
    @Test
    public void doesntProtectAgainstStubMutationByTransformers() {
        wm = new WireMockServer(wireMockConfig()
                .dynamicPort()
                .extensions(new ResponseDefinitionTransformer(){
                    @Override
                    public String getName() {
                        return "evil";
                    }

                    @Override
                    public ResponseDefinition transform(Request request, ResponseDefinition responseDefinition, FileSource files, Parameters parameters) {
                        responseDefinition.getTransformerParameters().clear();
                        responseDefinition.getTransformers();
                        return responseDefinition;
                    }
                }));
        wm.start();

        StubMapping stubMapping = wm.stubFor(get(urlEqualTo("/evil-mutation"))
                .willReturn(aResponse()
                .withHeader("MyHeader", "Initial")
                .withTransformerParameter("MyTransformerParam", "value")
                .withStatus(200)
                .withBody("Can you see MyHeader?")));

        assertThat(stubMapping.getResponse().getTransformerParameters().size(), is(1));

        new WireMockTestClient(wm.port()).get("/evil-mutation");

        assertThat(stubMapping.getResponse().getTransformerParameters().size(), is(0));

    }
```

not sure how bad is that though. Let me know if this would better be fixed by moving the copying to the line 67. Would prefer as a separate follow-up change though as this one has grown enough.